packages/application/src/requests.tsx

```diff
@@ -0,0 +1,137 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { showDialog, Dialog } from '@jupyterlab/apputils';
+import { URLExt } from '@jupyterlab/coreutils';
+import { ServerConnection } from '@jupyterlab/services';
+
+import * as React from 'react';
+
+import { ErrorDialogContent } from './ErrorDialogContent';
+
+const HTML_STATUS_SUCCESS = [200, 201];
+
+export class RequestHandler {
+  static serverError(response: any): Promise<Dialog.IResult<any>> {
+    const reason = response.reason ? response.reason : '';
+    const message = response.message ? response.message : '';
+    const timestamp = response.timestamp ? response.timestamp : '';
+    const traceback = response.traceback ? response.traceback : '';
+    const default_body = response.timestamp
+      ? 'Check the JupyterLab log for more details at ' + response.timestamp
+      : 'Check the JupyterLab log for more details';
+
+    return showDialog({
+      title: 'Error making request',
+      body:
+        reason || message ? (
+          <ErrorDialogContent
+            reason={reason}
+            message={message}
+            timestamp={timestamp}
+            traceback={traceback}
+            default_msg={default_body}
+          />
+        ) : (
+          <p>{default_body}</p>
+        ),
+      buttons: [Dialog.okButton()]
+    });
+  }
+
+  static server404(): Promise<Dialog.IResult<any>> {
+    return showDialog({
+      title: 'Error contacting server',
+      body: <p>Elyra service endpoint not found.</p>,
+      buttons: [Dialog.okButton()]
+    });
+  }
+
+  static async makeGetRequest(
+    requestExt: string,
+    displayWaitDialog: boolean
+  ): Promise<any> {
+    return this.makeServerRequest(
+      requestExt,
+      { method: 'GET' },
+      displayWaitDialog
+    );
+  }
+
+  static async makePostRequest(
+    requestExt: string,
+    requestBody: any,
+    displayWaitDialog: boolean
+  ): Promise<any> {
+    return this.makeServerRequest(
+      requestExt,
+      { method: 'POST', body: requestBody },
+      displayWaitDialog
+    );
+  }
+
+  static async makeServerRequest(
+    requestExt: string,
+    requestOptions: any,
+    displayWaitDialog: boolean
+  ): Promise<any> {
+    // use ServerConnection utility to make calls to Jupyter Based services
+    // which in this case are the in the extension installed by this package
+    const settings = ServerConnection.makeSettings();
+    const requestUrl = URLExt.join(settings.baseUrl, requestExt);
+
+    console.log(
+      'Sending a ' + requestOptions.method + ' request to ' + requestUrl
```

Comment:
more of a nit (or personal preference on my part). i see this (string concatenation) throughout. any particular reason for not using template literals instead? i tend to find them more readable. for example

```javascript
console.log(`Sending a ${requestOptions.method} request to ${requestUrl}`)
```

Reply:
This was old code, but I agree, I'll update it

Reply:
@vabarbosa Please definetlly educate the `backend engineers`